### PR TITLE
CLOUDP-193173: Add extra args to serverless tags updates

### PR DESCRIPTION
**Pull request by josvazg**
### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* ~~Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).~~
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: pkg/api/v1/atlasdeployment_types.go [hunk 1]
```diff
@@ -220,6 +220,12 @@ type ServerlessSpec struct {
 	// +kubebuilder:validation:MaxItems=50
 	// +optional
 	Tags []*TagSpec `json:"tags,omitempty"`
+
+	// (Serverless) Backup Options
+	BackupOptions ServerlessBackupOptions `json:"backupOptions,omitempty"`
+
+	// TerminationProtectionEnabled flag
+	TerminationProtectionEnabled *bool `json:"terminationProtectionEnabled,omitempty"`
```
Comment (igor-karpukhin) [inline pkg/api/v1/atlasdeployment_types.go:228]: You can set a default value here: `+kubebuilder:default:=false`

File: pkg/api/v1/atlasdeployment_types.go [hunk 2]
```diff
@@ -247,6 +253,10 @@ type TagSpec struct {
 	Value string `json:"value"`
 }
 
+type ServerlessBackupOptions struct {
+	ServerlessContinuousBackupEnabled *bool `json:"serverlessContinuousBackupEnabled,omitempty"`
```
Comment (igor-karpukhin) [inline pkg/api/v1/atlasdeployment_types.go:257]: Same here: `You can set a default value here: `+kubebuilder:default:=true`

File: config/crd/bases/atlas.mongodb.com_atlasdeployments.yaml [hunk 3]
```diff
@@ -665,6 +665,14 @@ spec:
               serverlessSpec:
                 description: Configuration for the serverless deployment API. https://www.mongodb.com/docs/atlas/reference/api/serverless-instances/
                 properties:
+                  backupOptions:
+                    description: (Serverless) Backup Options
```
Comment (igor-karpukhin) [inline config/crd/bases/atlas.mongodb.com_atlasdeployments.yaml:669]: note: redundant parentheses

File: pkg/controller/atlasdeployment/serverless_deployment.go [hunk 4]
```diff
@@ -89,3 +90,13 @@ func isTagsEqual(a []*mongodbatlas.Tag, c []*mongodbatlas.Tag) bool {
 	}
 	return false
 }
+
+func updateServerlessRequest(tags *[]*mongodbatlas.Tag, serverlessSpec *mdbv1.ServerlessSpec) *mongodbatlas.ServerlessUpdateRequestParams {
```
Comment (igor-karpukhin) [inline pkg/controller/atlasdeployment/serverless_deployment.go:94]: note: you can get rid of this method that only fills the struct and move the body to the place where it's called

**Review by igor-karpukhin — APPROVED** [review]
Looks good in general 👍 See notes

File: pkg/controller/atlasdeployment/serverless_deployment.go (hunk 4 above)
Comment (josvazg) [reply pkg/controller/atlasdeployment/serverless_deployment.go]: done

File: config/crd/bases/atlas.mongodb.com_atlasdeployments.yaml (hunk 3 above)
Comment (josvazg) [reply config/crd/bases/atlas.mongodb.com_atlasdeployments.yaml]: done

**Review by igor-karpukhin — APPROVED** [review]
Good job 👍

**Pull request merged**